apidump/apidump.go

```diff
@@ -121,6 +122,106 @@ type Args struct {
 	ParseTLSHandshakes bool
 }
 
+// TODO: either remove write-to-local-HAR-file completely,
+// or refactor into a separate class to avoid all the branching.
+type apidump struct {
+	*Args
+
+	backendSvc  akid.ServiceID
+	learnClient rest.LearnClient
+
+	startTime   time.Time
+	dumpSummary *Summary
+}
+
+// Start a new apidump session based on the given arguments.
+func newSession(args *Args) *apidump {
+	a := &apidump{
+		Args:      args,
+		startTime: time.Now(),
+	}
+	return a
+}
+
+// Is the target the Akita backend as expected, or a local HAR file?
+func (a *apidump) TargetIsRemote() bool {
+	return a.Out.AkitaURI != nil
+}
+
+// Lookup the service and create a learn client targeting it.
+func (a *apidump) LookupService() error {
+	if !a.TargetIsRemote() {
+		return nil
+	}
+	frontClient := rest.NewFrontClient(a.Domain, a.ClientID)
+	backendSvc, err := util.GetServiceIDByName(frontClient, a.Out.AkitaURI.ServiceName)
+	if err != nil {
+		return err
+	}
+	a.backendSvc = backendSvc
+	a.learnClient = rest.NewLearnClient(a.Domain, a.ClientID, backendSvc)
+	return nil
+}
+
+// Send the initial mesage to the backend indicating successful start
+func (a *apidump) SendInitialTelemetry() {
+	// The observed duration serves as a key for upsert, so
+	// it should be the same on the initial empty report indicating
+	// succesful startup, and the one sixty seconds later.
+	req := &kgxapi.PostClientPacketCaptureStatsRequest{
+		ObservedDurationInSeconds: a.StatsLogDelay,
+	}
+	a.SendTelemetry(req)
+}
```

Comment:
The old version of this had `req.PacketCountSummary = dumpSummary.FilterSummary.Summary(topNForSummary)`. Is this no longer needed/desired?

Reply:
The initial summary should be empty.  I puzzled about that for a while; the API is documented such that PacketCountSummary should be absent, so I was surprised to see the call there. 

> 	// If PacketCountSummary is absent, then this observation period has
>     // started but not yet concluded.
 
 This seems to work OK on the server side, but it is a behavior change.